src/TulsiGenerator/Bazel/tulsi/tulsi_aspects.bzl

```diff
@@ -51,33 +51,33 @@ _SUPPORTING_FILE_ATTRIBUTES = [
     'xibs',
 ]
 
-# Set of rules with implicit <label>.ipa IPA outputs.
+# List of rules with implicit <label>.ipa IPA outputs.
 # TODO(b/33050780): This is only used for the native rules and will be removed
 # in the future
-_IPA_GENERATING_RULES = set([
+_IPA_GENERATING_RULES = [
```

Comment:
The reason is that `set` and `depset` are the same data structure which is not a hashset but a [dag](https://docs.bazel.build/versions/master/skylark/depsets.html) and checks in them are linear. This confusion is basically the main reason why we're renaming sets to depsets.

These lists are relatively small and I think linear searches in them is not a bottleneck, but if it is you can use dicts that are proper hash tables with O(1) lookups: `IPA_GENERATING_RULES = {x: None for x in ...}`. We can introduce proper hash sets in the future but not before we finally remove `set()` for depsets.